src/telemetry.c

```diff
@@ -1016,6 +1090,55 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
         return ret;
 }
 
+
+/**
+ * Checks for id to have length = 32 characters
+ * and hexadecimal characters only.
+ *
+ * @param id A pointer to string to be checked
+ *
+ * @return 0 on success, 1 on failure
+ *
+ */
+static int validate_event_id(char *id)
+{
+       int rc = 0;
+       const char alphab[] = EVENT_ID_ALPHAB;
+
+       // 32 - 32 = 0
+       if((rc = (int)(strlen(id) - EVENT_ID_LEN)) != 0) {
+               return 1;
+       }
+
+       // verify alphabet
+       // strspn checks for characters in alphab
+       if(id[strspn(id, alphab)] != 0) {
+               return 1;
+       }
+
+       return rc;
+}
+
+
+int tm_set_event_id(struct telem_ref *t_ref, char *event_id)
+{
+        int status = 0;
+
+        if(!(status = validate_event_id(event_id))) {
+                 // free default id before overriding
+                 free(t_ref->record->headers[TM_EVENT_ID]);
```

Comment:
Check for t_ref == NULL and t_ref->record == NULL and t_ref->record->headers == NULL before this deref.

Reply:
I agree. It's better to be defensive here, since `tm_set_event_id` is the new API function.